073-extend-feature-gates-to-all-operators.md

```diff
@@ -0,0 +1,114 @@
+The same feature gates will be defined and handled by every operator.
+However, if the operator doesn't have any use for the feature gate, it can simply ignore it in its code.
+
+When the User and Topic Operators are deployed through the Cluster Operator as part of the Kafka cluster, the user will configure the feature gates as today using the `STRIMZI_FEATURE_GATES` environment variable.
+And the Cluster Operator will then automatically pass the environment variable to the User and Topic Operators when deploying them.
+When the User and Topic Operators are deployed as standalone, user can configure the `STRIMZI_FEATURE_GATES` environment variable directly in their deployments.
+
+### Trade-offs and limitations
+
+This design has several trade-offs and limitations: 
+* When a feature gate is enabled or disabled in the Cluster Operator, it will cause rolling update of all the User and Topic Operators managed by it.
+  This will happen even if they in reality don't use given feature gate in their code and simply ignore it.
+  That said, changing the feature gates configuration is not something what is expected to be done very often.
+* The centrally managed feature gate configuration will also apply to all Kafka clusters managed by given Cluster Operator.
+  It cannot be applied to individual Kafka clusters which can be seen as a limitation.
+  But it does not differ significantly from the current state.
+
+### Advantages
+
+It has also aspects that can be considered as advantages:
+* The configuration of the feature gates is unchanged compared to how it works today.
+  It also maintains the simplicity of the configuration.
+  The user needs to set only one environment variable in the Cluster Operator to configure all feature gates.
+  This also means that we do not need any changes to our system tests as they can keep using the Feature Gates in the same way as they do today.
+* Having a consistency and clarity how the operators are configured simplifies support as there are less variables that need to be understood before helping our users.
+* While the implementation is different, the proposed behavior mirrors the current behavior:
+    * The feature gates apply to all operands (i.e. to all User and Topic Operators managed by the same Cluster Operator)
+    * It follows the behavior of the `UnidirectionalTopicOperator` feature gate that was also configured centrally and changing it caused rolling update of the Topic Operator(s).
+* Given that one of the goals of the feature gates is to allow users easily test new features before introducing them to everyone when they are enabled by default, the central configuration makes sure the feature gate is properly tested.
+  It for example helps to avoid situation when a user would by mistake test the `UseServerSideApply` feature gate only in the Cluster Operator, but not in the other operators.
```

Comment:
```suggestion
  For example, it helps to avoid situation when a user would test the `UseServerSideApply` feature gate only in the Cluster Operator by mistake, but not in the other operators.
```